packages/office-ui-fabric-react/src/components/Nav/Nav.tsx

```diff
@@ -216,9 +216,10 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
   private _isLinkSelected(link: INavLink): boolean {
     // if caller passes in selectedKey, use it as first choice or
     // if current state.selectedKey (from addressbar) is match to the link
-    if (this.props.selectedKey !== undefined && link.key === this.props.selectedKey ||
-        this.state.selectedKey !== undefined && link.key === this.state.selectedKey) {
-      return true;
+    if (this.props.selectedKey !== undefined ) {
+      return link.key === this.props.selectedKey;
+    } else if (this.state.selectedKey !== undefined) {
+      return link.key === this.state.selectedKey;
```

Comment:
private _onNavAnchorLinkClicked(link: INavLink, ev: React.MouseEvent<HTMLElement>): void {
    if (this.props.onLinkClick) {
      this.props.onLinkClick(ev, link);
    }

    this.setState({ selectedKey: link.key });
  }
basically this.state.selectedKey is the address bar value before browser potentially redirect or resolve?